utils/hct/hctbuild.cmd

```diff
@@ -55,173 +43,185 @@ set DXILCONV=ON
-  shift /1
+  shift /1 & goto :parse_args
 )
 if "%1"=="-x86" (
   set BUILD_ARCH=Win32
-  shift /1
+  shift /1 & goto :parse_args
 )
 if "%1"=="-x64" (
   set BUILD_ARCH=x64
-  shift /1
+  shift /1 & goto :parse_args
 )
 if /i "%1"=="-arm" (
   set BUILD_ARCH=ARM
-  shift /1
+  shift /1 & goto :parse_args
 )
 if /i "%1"=="-arm64" (
   set BUILD_ARCH=ARM64
-  shift /1
+  shift /1 & goto :parse_args
 )
 if /i "%1"=="-arm64ec" (
   set BUILD_ARCH=ARM64EC
-  shift /1
+  shift /1 & goto :parse_args
 )
 if /i "%1"=="-Debug" (
   set BUILD_CONFIG=Debug
-  shift /1
+  shift /1 & goto :parse_args
 )
 if /i "%1"=="-Release" (
   set BUILD_CONFIG=Release
-  shift /1
+  shift /1 & goto :parse_args
 )
 if "%1"=="-vs2017" (
   set BUILD_GENERATOR=Visual Studio 15 2017
   set BUILD_VS_VER=2017
-  shift /1
+  shift /1 & goto :parse_args
 )
 if "%1"=="-vs2019" (
-  shift /1
+  shift /1 & goto :parse_args
 )
-
 if "%1"=="-tblgen" (
   if "%~2" == "" (
     echo Missing path argument after -tblgen.
     exit /b
   ) 
   set "BUILD_TBLGEN_PATH=%~2"
   shift /1
-  shift /1
+  shift /1 & goto :parse_args
 )
-
 if "%1"=="-dont-speak" (
-  shift /1
+  shift /1 & goto :parse_args
 )
-
 if "%1"=="-speak-up" (
   set SPEAK=1
-  shift /1
+  shift /1 & goto :parse_args
 )
-
 if "%1"=="-no-parallel" (
   set PARALLEL_OPT=
-  shift /1
+  shift /1 & goto :parse_args
 )
-
 if "%1"=="-no-dxilconv" (
   set DXILCONV=OFF
-  shift /1
+  shift /1 & goto :parse_args
 )
-
 if "%1"=="-dxc-cmake-extra-args" (
   set CMAKE_OPTS=%CMAKE_OPTS% %~2
   shift /1
-  shift /1
+  shift /1 & goto :parse_args
 )
-
 if "%1"=="-dxc-cmake-begins-include" (
   set "CMAKE_OPTS=%CMAKE_OPTS% -DDXC_CMAKE_BEGINS_INCLUDE=%~2"
   shift /1
-  shift /1
+  shift /1 & goto :parse_args
 )
-
 if "%1"=="-dxc-cmake-ends-include" (
   set "CMAKE_OPTS=%CMAKE_OPTS% -DDXC_CMAKE_ENDS_INCLUDE=%~2"
   shift /1
-  shift /1
+  shift /1 & goto :parse_args
 )
-
 if "%1"=="-dxc-cmake" (
   set "CMAKE_PATH=%~2"
   shift /1
-  shift /1
+  shift /1 & goto :parse_args
 )
-
 if "%1"=="-dxc-cmake-system-version" (
   set DXC_CMAKE_SYSTEM_VERSION=%~2
   shift /1
-  shift /1
+  shift /1 & goto :parse_args
 )
 
 if "%1"=="-show-cmake-log" (
   set SHOW_CMAKE_LOG=1
-  shift /1
+  shift /1 & goto :parse_args
 )  
-
-if "%CMAKE_PATH%"=="" (
-  where cmake.exe 1>nul 2>nul
-  if errorlevel 1 (
-    echo Unable to find cmake.exe on the path.
-    echo cmake 3.4 is available from https://cmake.org/files/v3.4/cmake-3.4.0-win32-x86.exe
-    exit /b 1
-  )
-  set CMAKE_PATH=cmake
-)
-
 rem Begin SPIRV change
 if "%1"=="-spirv" (
   echo SPIR-V codegen is enabled.
   set SPIRV=ON
-  shift /1
+  shift /1 & goto :parse_args
 )
 if "%1"=="-spirvtest" (
   echo Building SPIR-V tests is enabled.
   set SPV_TEST=ON
-  shift /1
+  shift /1 & goto :parse_args
 )
 rem End SPIRV change
+if "%1"=="-ninja" (
+  set BUILD_GENERATOR=Ninja
+  shift /1 & goto :parse_args
+)
+if "%1" NEQ "" ( 
+    echo Unrecognized argument: %1
+    exit /b 1
+)
```

Comment:
Very welcome erroneous use checking! I think I would have forgotten this.

just one tiny improvement, I think a `goto :showhelp` would be even more helpful here.

Reply:
I though about it. `goto :showhelp` would mask the exit error code and it should return error in this case. I did want to add error code handling to `:showhelp`.